platform/logging/LoggerService.ts

```diff
@@ -0,0 +1,26 @@
+import { Observable, Subject } from 'rxjs';
+
+import { MutableLoggerFactory } from './LoggerFactory';
+import { LoggingConfig } from './LoggingConfig';
+
+// The `LoggerManager` is responsible for maintaining the log config
+// subscription and pushing updates the the mutable logger.
+
+export class LoggerService {
+  private readonly stop$ = new Subject();
+
+  constructor(private readonly loggingFactory: MutableLoggerFactory) {}
+
+  upgrade(config$: Observable<LoggingConfig>) {
+    config$.takeUntil(this.stop$).subscribe({
```

Comment:
I don't close `LoggingFactory` in `complete` handler anymore because of the fact that it's called when `RawLoggingService` is closed that happens _before_ we stop other subsystems (eg. `plugins`).